subprojects/core/src/main/java/org/gradle/internal/service/scopes/BuildScopeServices.java

```diff
@@ -416,6 +425,44 @@ protected ITaskFactory createITaskFactory(Instantiator instantiator, TaskClassIn
             new TaskFactory());
     }
 
+    protected GroovyScriptClasspathProvider createGroovyScriptClasspathProvider(
+        ClassLoaderScopeRegistry classLoaderScopeRegistry,
+        DependencyManagementServices dependencyManagementServices,
+        FileResolver fileResolver,
+        FileCollectionFactory fileCollectionFactory,
+        DependencyMetaDataProvider dependencyMetaDataProvider,
+        DependencyFactory dependencyFactory
+    ) {
+        DependencyResolutionServices resolutionServices = dependencyManagementServices.create(
+            fileResolver,
+            fileCollectionFactory,
+            dependencyMetaDataProvider,
+            new UnknownProjectFinder("Some unknown project"),
+            RootScriptDomainObjectContext.PLUGINS
+        );
+        return new GroovyScriptClasspathProvider(
+            classLoaderScopeRegistry.getCoreAndPluginsScope(),
+            () -> {
+                String version = System.getProperty("org.gradle.api.version");
```

Comment:
We should qualify what API version this is. I'd go with `org.gradle.api.target-version` or something.

Reply:
Yeah, I think we should qualify the property name. It feels like `target` doesn't really capture it. For me, `target` sounds more about the target runtime. How about `org.gradle.api.source-version`, since this is the version of the Gradle API we use for sources.

Reply:
Renamed.